includes/class-mailer.php

```diff
@@ -148,8 +148,12 @@ public static function new_follower( $notification ) {
 	 * @param int   $user_id  The id of the local blog-user.
 	 */
 	public static function direct_message( $activity, $user_id ) {
-		// Check if Activity is public or not.
-		if ( is_activity_public( $activity ) ) {
+		if (
+			is_activity_public( $activity ) ||
+			// Only accept messages that have the user in the "to" field.
+			empty( $activity['to'] ) ||
+			! in_array( \get_author_posts_url( $user_id ), (array) $activity['to'], true )
```

Comment:
it can also be the `?author=<id>` URL I think we should use `User->get_id()` instead

Reply:
Ok, I'll change it!